compiler/rustc_hir_typeck/src/fn_ctxt/adjust_fulfillment_errors.rs

```diff
@@ -474,6 +474,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             expr,
         )?;
 
+        if self.tcx.is_trait_alias(obligation.impl_def_id) {
```

Comment:
Yeah this works -- or we could synthesize a trait ref using the identity substitutions of the trait alias def id. 

But this works for now.

Reply:
Okay, I'll do that.  Won't be for a few hours though.

Reply:
Done!

@rustbot ready